fix: actually erase old evodb data on db migration

UdjinM6 opened the pull request:
## Issue being fixed or feature implemented
The old evodb data wasn't dropped on db migration really. `Erase(<db_key_prefix>)` does nothing.

## What was done?
Loop through the old data and drop it in batches per db key. Do this both for nodes that are doing migration for the first time and for nodes that did migration in the past already.

## How Has This Been Tested?
Running different versions on testnet
```
# reindex with 18.2.2 till block 850000 (pre-v19 block)
$ du -hd1 ~/.dashcore/testnet3/evodb/
276M	.dashcore/testnet3/evodb

# continue with develop, migration just finished, keep syncing till current tip, block 901000+
$ du -hd1 ~/.dashcore/testnet3/evodb/
469M	.dashcore/testnet3/evodb

# continue with this PR, start at current tip, "migration already done. cleaned old data."
$ du -hd1 ~/.dashcore/testnet3/evodb/
302M	.dashcore/testnet3/evodb
```

## Breaking Changes
Should be none.

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

knst commented on src/evo/deterministicmns.cpp:
```diff
@@ -1181,6 +1181,31 @@ void CDeterministicMNManager::CleanupCache(int nHeight)
+        }
+        if (erased) {
+            db.WriteBatch(batch);
+            LogPrintf("CDeterministicMNManager::%s -- done cleaning old data for %s\n", __func__, db_key);
```
maybe better to have a log before `WriteBatch` for case if it would be failed - at least something would be in logs to point a place of failure.

knst commented on src/evo/deterministicmns.cpp:
```diff
@@ -1181,6 +1181,31 @@ void CDeterministicMNManager::CleanupCache(int nHeight)
+    for(const auto& db_key : db_keys) {
+        CDBBatch batch(db);
+        auto it = std::unique_ptr<CDBIterator>(db.NewIterator());
+        auto firstKey = std::make_pair(db_key, uint256());
```
```suggestion
        std::pair firstKey{db_key, uint256()};
```

knst commented on src/evo/deterministicmns.cpp:
```diff
@@ -1181,6 +1181,31 @@ void CDeterministicMNManager::CleanupCache(int nHeight)
+{
+    bool erased{false};
+    for(const auto& db_key : db_keys) {
+        CDBBatch batch(db);
```
```suggestion
        CDBBatch batch{db};
```

knst reviewed (commented):
check some nits. Logic seems fine... Probably ok

ogabrielides commented:
@UdjinM6 

> Erase(<db_key>) does nothing.

Why ?

`  bool Erase(const K& key, bool fSync = false)
    {
        CDBBatch batch(*this);
        batch.Erase(key);
        return WriteBatch(batch, fSync);
    }`

UdjinM6 commented:
> @UdjinM6
> 
> > Erase(<db_key>) does nothing.
> 
> Why ?
> 
> ` bool Erase(const K& key, bool fSync = false) { CDBBatch batch(*this); batch.Erase(key); return WriteBatch(batch, fSync); }`

I meant prefix string there. Fixed description and added f7a97fc.

UdjinM6 commented on src/evo/deterministicmns.cpp:
```diff
@@ -1181,6 +1181,31 @@ void CDeterministicMNManager::CleanupCache(int nHeight)
+    bool erased{false};
+    for(const auto& db_key : db_keys) {
+        CDBBatch batch(db);
+        auto it = std::unique_ptr<CDBIterator>(db.NewIterator());
```
? I need `it` to loop through

knst replied:
sorry, typo:
```
+        std::unique_ptr<CDBIterator> it{db.NewIterator()};
```

knst reviewed (approved):
LGTM

ogabrielides reviewed (approved):
ACK

PastaPastaPasta reviewed (approved):
utACK for squash merge
